prow/plank/controller.go

```diff
@@ -291,6 +291,11 @@ func (c *Controller) SyncMetrics() {
 func (c *Controller) terminateDupes(pjs []prowapi.ProwJob, pm map[string]coreapi.Pod) error {
 	log := c.log.WithField("aborter", "pod")
 	return pjutil.TerminateOlderJobs(c.kc, log, pjs, func(toCancel prowapi.ProwJob) error {
+		// Skip report for terminated dup jobs
+		if err := c.setPreviousReportState(toCancel); err != nil {
+			c.log.WithFields(pjutil.ProwJobFields(&toCancel)).WithError(err).Error("Failed to patch PrevReportStates")
+		}
```

Comment:
This cleanup function is called before the state is actually updated:
https://github.com/kubernetes/test-infra/blob/5839ecdd29c8627c305a1860efd908136bc48c2e/prow/pjutil/abort.go#L100
Also note that since the `setPreviousReportState` function fetches the most recent ProwJob, the updated state would have to be applied to the cluster before the `setPreviousReportState` is called, but that would potentially trigger crier to react to the state change before `setPreviousReportState` can be updated. 
I think the `PrevReportStates` map needs to be updated at the same time the state is to avoid a race with `crier`.

Reply:
good point, I'll move it inside `TerminateOlderJobs`

Reply:
done